### ISSUE TITLE
www.facebook.com (the desktop site) doesn't work

### ISSUE THREAD

**Issue by akorn** [tobykurien/WebApps#222]
Instead of m.facebook.com, where as of about yesterday, the Messenger icon in the navigation bar stopped working (I assume Facebook wants to force me to install their Messenger app, which I will not do), I tried to add a webapp for www.facebook.com.

First I set my user agent to `Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.26 Safari/537.36 OPR/74.0.3862.0 (Edition developer)` in an attempt to force the site to treat my browser as a desktop client.

I then chose "new webapp", entered www.facebook.com as the URL, and tapped "OK". All that happened was that a toast notification with "URL: https://www.facebook.com/" was displayed briefly.

I tried with various combinations of different URL (e.g. https://facebook.com/), allowing 3rd party requests, allowing redirects, deselecting "open in new webapp" (what does that even do), but the result was, sadly, always the same.

**Comment by eapo** [comment]
Would be a good solution to add a **new button** next to the `menu_image` as `menu_script` and _toggle_ a predefined script per WebApp to be able to use `document.getElementsByClassName('n5hh3s10')[0].style.display = "none";` and `document.getElementsByClassName('n5hh3s10')[0].style.display = "block";` for the _messenger dot com_'s desktop view.
**Who can realize that?**
Please 👍!

**Comment by creativetrendsapps** [comment]
> Would be a good solution to add a **new button** next to the `menu_image` as `menu_script` and _toggle_ a predefined script per WebApp to be able to use `document.getElementsByClassName('n5hh3s10')[0].style.display = "none";` and `document.getElementsByClassName('n5hh3s10')[0].style.display = "block";` for the _messenger dot com_'s desktop view.
> **Who can realize that?**
> Please 👍! 

Why are you hiding the element and then displaying as block? How would the block actually work?

**Comment by eapo** [comment]
> 
> 
> > Would be a good solution to add a **new button** next to the `menu_image` as `menu_script` and _toggle_ a predefined script per WebApp to be able to use `document.getElementsByClassName('n5hh3s10')[0].style.display = "none";` and `document.getElementsByClassName('n5hh3s10')[0].style.display = "block";` for the _messenger dot com_'s desktop view.
> > **Who can realize that?**
> > Please 👍!
> 
> Why are you hiding the element and then displaying as block? How would the block actually work?

You are right, originally it's `flex`. Here are the screenshots with &hellip;`.style.display` toggle

![document.getElementsByClassName('n5hh3s10')[0].style.display = "flex";](https://i.imgur.com/RLKBbuv.png ".display = 'flex';")
![document.getElementsByClassName('n5hh3s10')[0].style.display = "none";](https://i.imgur.com/d1kdIg3.png ".display = 'none';")

**Comment by creativetrendsapps** [comment]
> > > Would be a good solution to add a **new button** next to the `menu_image` as `menu_script` and _toggle_ a predefined script per WebApp to be able to use `document.getElementsByClassName('n5hh3s10')[0].style.display = "none";` and `document.getElementsByClassName('n5hh3s10')[0].style.display = "block";` for the _messenger dot com_'s desktop view.
> > > **Who can realize that?**
> > > Please 👍!
> > 
> > 
> > Why are you hiding the element and then displaying as block? How would the block actually work?
> 
> You are right, originally it's `flex`. Here are the screenshots with …`.style.display` toggle
> 
> ![document.getElementsByClassName('n5hh3s10')[0].style.display = "flex";](https://camo.githubusercontent.com/11d1d9b6dada2a4a0f9aa087464c8df0875d05120a29991539a5e8b44cb7ea33/68747470733a2f2f692e696d6775722e636f6d2f524c4b426275762e706e67)
> ![document.getElementsByClassName('n5hh3s10')[0].style.display = "none";](https://camo.githubusercontent.com/b3c58a0fa0385022f5c950ed5ed22c12736afe7a58fc12c697b566b89622217e/68747470733a2f2f692e696d6775722e636f6d2f64316b644967332e706e67)

This works for me:

/*sfb*/

.wkznzc2l {
    display: none !important;
}

.kuivcneq {
    display: none !important;
}

.bafdgad4 {
    display: none !important;
}

.aov4n071.cxmmr5t8.bi6gxh9e.hcukyx3x.jb3vyjys.hv4rvrfc.qt6c0cv9.dati1w0a {
    display: none !important;
}


.rj1gh0hx {
    max-width: -webkit-fill-available;
}

.j83agx80 {
    max-width: -webkit-fill-available;
}

also for web settings:  mWebView.setInitialScale(160);

**Comment by tobykurien** [comment]
As far as I know, it is not possible to inject code (JS/CSS) into a webview from an app (it is a big security risk, e.g. grabbing your credit card details).

**Comment by creativetrendsapps** [comment]
> As far as I know, it is not possible to inject code (JS/CSS) into a webview from an app (it is a big security risk, e.g. grabbing your credit card details).

This is false. Injecting CSS is not pulling out data, it's overriding styles. Every wrapper you use (including Frost) injects CSS.
